### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by TheRealChriAnd**
## Purpose
  - Fixed: Ready sometimes not set for new players joining the lobby
  - Fixed: A name of 24 characters will cause the lobby to crash
  - Fixed: Teams are not rebalanced in the lobby when a player leaves
  - Fixed: The server keeps waiting for a player that disconnected during loading
  - Fixed: The chat is not cleared when rejoining a lobby
  - Fixed: The number of players label in the lobby sometimes shows an incorrect amount.
  - Fixed: Loading the world takes more than 5 Seconds in debug which causes the ping system to disconnect.
  - Fixed: The max number of players is now displayed in the server list

## Testing
- [x] Tested in Sandbox
- [x] Tested in Multiplayer with 2 clients
- [ ] Tested in Multiplayer with more than 2 clients
- [x] Tested in Benchmark

### CODE REVIEW

File: CrazyCanvas/Source/Lobby/PlayerManagerServer.cpp [hunk 1]
```diff
@@ -231,6 +231,23 @@ void PlayerManagerServer::HandlePlayerLeftServer(LambdaEngine::IClient* pClient)
 	{
 		SetPlayerHost(&(s_Players.begin()->second));
 	}
+
+	if (s_Players.size() < 2 || ServerState::GetState() != SERVER_STATE_LOBBY)
+		return;
+
+	TArray<const Player*> pPlayersTeam0;
+	TArray<const Player*> pPlayersTeam1;
```
Comment (TimTheToad) [inline CrazyCanvas/Source/Lobby/PlayerManagerServer.cpp]: thsese shouldn't be pPlayersTeamX, but playersTeamx. Right?

**Review by TimTheToad — CHANGES_REQUESTED** [review]
Looks very good, Nice =)

**Review by Vonsen — APPROVED** [review]
Looks good to me

**Review by TimTheToad — APPROVED** [review]
dutti!

**Pull request merged**